### PR TITLE
 Convert `packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js` and `packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.test.js`  to RTL

### LINKED ISSUE #48253: Convert all remaining enzyme tests to React Testing Library
Enzyme doesn't officially support react 18, or even react 17 yet...

- [x] #49586
- [x] #49063
- [x] #48292
- [x] #49067
- [x] #49587
- [x] #49250
- [x] #49588
- [x] #49589
- [x] #48283
- [x] #48886
- [x] Convert `packages/grafana-ui/src/components/Logs/LogMessageAnsi.test.tsx` to RTL
- [x] #55030
- [x] https://github.com/grafana/grafana/pull/48982
- [ ] #57884
- [x] #48260
- [x] #49707
- [x] #49507
- [x] #48290
- [x] Convert `packages/grafana-ui/src/slate-plugins/braces.test.tsx` to RTL
- [x] Convert `packages/grafana-ui/src/slate-plugins/clear.test.tsx` to RTL
- [x] Convert `packages/grafana-ui/src/slate-plugins/runner.test.tsx` to RTL
- [x] Convert `packages/grafana-ui/src/slate-plugins/suggestions.test.tsx` to RTL
- [x] #50348
- [x] #50349
- [x] #50402
- [x] #50577
- [x] #57885
- [x] #57886
- [x] #51227
- [x] #49992
- [x] #57887
- [x] Convert `packages/jaeger-ui-components/src/TraceTimelineViewer/ReferencesButton.test.js` to RTL
- [x] #49107 
- [x] #49119
- [x] #51258
- [x] #51266
- [x] #51365
- [x] https://github.com/grafana/grafana/pull/49148
- [x] #51299
- [x] https://github.com/grafana/grafana/pull/49148
- [x] #56017
- [x] https://github.com/grafana/grafana/issues/51688
- [x] #56178
- [x] #50041
- [x] #51300
- [x] #57888
- [x] #51741
- [x] #57889
- [x] #57890
- [x] #56275
- [x] Convert `packages/jaeger-ui-components/src/common/CopyIcon.test.js` to RTL
- [x] #49827
- [x] #49994
- [x] #50332
- [x] #49590
- [x] #51146
- [x] #51625
- [x] #49835
- [x] #56800
- [x] Convert `public/app/features/alerting/AlertRuleList.test.tsx` to RTL
- [x] #53952
- [x] #50668
- [x] #49591
- [x] #49435
- [x] #49430
- [x] #49592
- [x] #49593
- [x] #49594
- [x] #50251
- [x] Convert `public/app/features/datasources/DataSourceDashboards.test.tsx` to RTL https://github.com/grafana/grafana/pull/51294
- [x] #51147
- [x] #49595
- [x] #49596
- [x] #56790
- [x] https://github.com/grafana/grafana/issues/52712
- [x] ~Convert `public/app/features/explore/RichHistory/RichHistoryStarredTab.test.tsx` to RTL~ https://github.com/grafana/grafana/pull/49492
- [x] ~Convert `public/app/features/explore/SecondaryActions.test.tsx` to RTL~ https://github.com/grafana/grafana/pull/48745
- [x] https://github.com/grafana/grafana/pull/48404
- [x] #49597
- [x] #50673
- [x] #49598
- [x] #49599
- [x] #49299 
- [x] https://github.com/grafana/grafana/issues/49399
- [x] https://github.com/grafana/grafana/issues/49467
- [x] https://github.com/grafana/grafana/issues/49808
- [x] https://github.com/grafana/grafana/issues/49914
- [x] https://github.com/grafana/grafana/issues/49974
- [x] https://github.com/grafana/grafana/issues/49975
- [x] #48257
- [x] #48256
- [x] #48255
- [x] #49403
- [x] #49402 
- [x] #49404
- [x] https://github.com/grafana/grafana/issues/52713
- [x] #51154
- [x] #49406 
- [x] #54331
- [x] https://github.com/grafana/grafana/issues/52714
- [x] https://github.com/grafana/grafana/issues/52715
- [x] https://github.com/grafana/grafana/issues/52716
- [x] https://github.com/grafana/grafana/issues/52717
- [x] https://github.com/grafana/grafana/issues/52718
- [x] #53958
- [x] #48254

### PR DESCRIPTION
**Pull request by xrisaD**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

For #48253

**Special notes for your reviewer**:

I added a function called `getRow ` at the `TextList` because:

When we test the `TextList` component we use as `data` a list of  dicts at the `TextList.test.js`, while at the  `TextList.test.js`the warnings we are passing is a list of strings. The existing solution (return the row) worked when having a list of strings but didn't work when having a list of dicts. I was getting the following error:

` Error: Uncaught [Error: Objects are not valid as a React child (found: object with keys {key, value}). If you meant to render a collection of children, use an array instead.]`

### CODE REVIEW

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js [hunk 1]
```diff
@@ -33,24 +33,26 @@ describe('<AccordianText>', () => {
   };
 
   beforeEach(() => {
-    wrapper = mount(<AccordianText {...props} />);
+    const r = render(<AccordianText {...props} />);
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js:36]: i'd just move the `render` function out of the `beforeEach` and into each test. then you don't need `rerender` at all.

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js [hunk 2]
```diff
@@ -33,24 +33,26 @@ describe('<AccordianText>', () => {
...
-    const content = wrapper.find(TextList);
-    expect(content.length).toBe(1);
-    expect(content.prop('data')).toBe(warnings);
+    props.isOpen = true;
+    rerender(<AccordianText {...props} />);
+    expect(screen.getAllByTestId('TextList')).toHaveLength(1);
+    const { getByText } = within(screen.getByTestId('TextList'));
+    warnings.forEach((warning) => {
+      expect(getByText(warning)).toBeInTheDocument();
+    });
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js:56]: let's just keep it simple:

```suggestion
    props.isOpen = true;
    render(<AccordianText {...props} />);
    warnings.forEach((warning) => {
      expect(screen.getByText(warning)).toBeInTheDocument();
    });
```

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.tsx [hunk 3]
```diff
@@ -86,7 +86,12 @@ export default function AccordianText(props: AccordianTextProps) {
   const styles = useStyles2(getStyles);
   return (
     <div className={className || ''}>
-      <div className={cx(styles.header, headerClassName)} {...headerProps} data-test-id="AccordianText--header">
+      <div
+        className={cx(styles.header, headerClassName)}
+        {...headerProps}
+        data-testid="AccordianText--header"
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.tsx:92]: same as before, `data-testid` is the correct format so i'd remove the `data-test-id` one and fix anything that needs fixing 🙂

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.test.js [hunk 4]
```diff
@@ -26,16 +26,15 @@ describe('<TextList>', () => {
   ];
 
   beforeEach(() => {
-    wrapper = shallow(<TextList data={data} />);
+    render(<TextList data={data} />);
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.test.js:29]: i'd move the render out of the `beforeEach` and into each test

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.test.js [hunk 5]
```diff
@@ -26,16 +26,15 @@ describe('<TextList>', () => {
...
   });
 
   it('renders without exploding', () => {
-    expect(wrapper).toBeDefined();
-    expect(wrapper.find('[data-test-id="TextList"]').length).toBe(1);
+    expect(screen).toBeDefined();
+    expect(screen.getAllByTestId('TextList')).toHaveLength(1);
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.test.js:34]:
```suggestion
    expect(() => render(<TextList data={data} />)).not.toThrow();
```

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js [hunk 6]
```diff
@@ -33,24 +33,26 @@ describe('<AccordianText>', () => {
...
   beforeEach(() => {
-    wrapper = mount(<AccordianText {...props} />);
+    const r = render(<AccordianText {...props} />);
+    rerender = r.rerender;
   });
 
   it('renders without exploding', () => {
-    expect(wrapper).toBeDefined();
-    expect(wrapper.exists()).toBe(true);
+    expect(screen).toBeDefined();
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianText.test.js:41]:
```suggestion
    expect(() => render(<AccordianText {...props} />)).not.toThrow();
```

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.tsx [hunk 7]
```diff
@@ -40,6 +40,16 @@ const getStyles = () => {
...
+const getRow = (row) => {
+  if (typeof row === 'string') {
+    return row;
+  } else {
+    // dict
+    Object.entries(row).map(([_, value]) => {
+      return { value };
+    });
+  }
+};
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/TextList.tsx:52]: can you add some info for why this new function is needed? 🤔

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
great start! 🙌

**Comment by xrisaD** [comment]
Why I added the `getRow` function:

When we test the `TextList` component we use as `data` a list of  dicts at the `TextList.test.js`, while at the  `TextList.test.js`the warnings we are passing is a list of strings. The existing solution (return the row) worked when having a list of strings but didn't work when having a list of dicts. I was getting the following error:

` Error: Uncaught [Error: Objects are not valid as a React child (found: object with keys {key, value}). If you meant to render a collection of children, use an array instead.]`


@ashharrison90 Probably there could be a better solution?

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
@xrisaD ah i see...

looking at the types we've defined for data in `TextList`, it should be an array of strings. so i would say we can remove the `getRow` function and change the `data` in the test file to be an array of strings instead, e.g. `const data = ['client', 'mos-def'];`

once you've done that and merged with main i think this looks great 👌

**Comment by xrisaD** [comment]
@ashharrison90 ok, I made these changes ✔️ 😄

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌

**Pull request merged**